### PR TITLE
Cleanup IndexResolver.buildIndices slightly

### PR DESCRIPTION
**Pull request by original-brownbear**
No need to check the unmapped array for membership, especially not with itself. The field caps response will not put an index into both unmapped and mapped ever, that woudl be an obvious bug. Also some minor drying up via collection tools we already have in place.
This should save a considerable amount of memory for not having to copy the unmapped fields array.

### CODE REVIEW

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java
Comment (original-brownbear) [inline x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java:656]: This was computing the size wrong, `newLInkedHashMapWithExpected` size already accounts for the load factor.

**Review by costin — APPROVED** [review]
The change looks alright to me - I'd wait for Andrei's approval though just to be sure.

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java
Comment (original-brownbear) [reply x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java:656]: The bug is not in `CollectionsUtils.mapSize` :), it's the fact that we apply both `CollectionsUtils.mapSize` **and then also** use `Map.capacity` in `newLinkedHashMapWithExpectedSize` so we oversize the map :)

**Review by astefan — APPROVED** [review]
LGTM. Thank you  @original-brownbear

**Comment by original-brownbear** [comment]
Thanks both!

**Pull request merged**
